runtime/JavaScript/src/antlr4/TokenStream.d.ts

```diff
@@ -1,16 +1,26 @@
-import {Interval} from "./misc";
-import {Token} from "./Token";
+/* Copyright (c) 2012-2022 The ANTLR Project. All rights reserved.
+ * Use of this file is governed by the BSD 3-clause license that
+ * can be found in the LICENSE.txt file in the project root.
+ */
+
+import Token from "./Token.js";
+import TokenSource from "./TokenSource.js";
+import Interval from "./misc/Interval.js";
 
 export declare class TokenStream {
+    public tokenSource: TokenSource;
+    public index: number;
+    public get size(): number;
```

Comment:
using a getter is left to the implementation, from a contract pov it's a field

Reply:
It's not. A single getter makes this member readonly (and shows there's potentially more processing than just a single field). Why hiding such information, if the language allows that?